integration/benchmark/cmd/ci/main.go

```diff
@@ -163,57 +168,76 @@ func main() {
 	if err != nil {
 		log.Fatal().Err(err).Msgf("unable to init loader")
 	}
-	lg.Start()
-
-	testStartTime := time.Now()
-	time.Sleep(loadCase.duration)
-	lg.Stop()
-	testEndTime := time.Now()
-
-	dataSlices := calculateTpsSlices(
-		testStartTime,
-		testEndTime,
-		*leadTime,
-		*sliceSize,
-		gitSha,
-		goVersion,
-		osVersion,
-		lg)
-	prepareDataForBigQuery(dataSlices, *ciFlag)
-}
 
-func calculateTpsSlices(start, end time.Time, leadTime, sliceTime, commit, goVersion, osVersion string, lg *benchmark.ContLoadGenerator) []dataSlice {
-	//remove the lead time on both start and end, this should remove spin-up and spin-down times
-	leadDuration, _ := time.ParseDuration(leadTime)
-	endTime := end.Add(-1 * leadDuration)
-	sliceDuration, _ := time.ParseDuration(sliceTime)
+	// run load
+	lg.Start()
 
-	slices := make([]dataSlice, 0)
+	// prepare data slices
+	dataSlices := make([]dataSlice, 0)
```

Comment:
This seems wrong.  `slice = append(slices, slice)` in line 236 won't modify this copy of the dataSlices variable.  As a result, your prepareDataForBigQuery call in the next line will always end up with an empty slice.

you probably want something more like this

```
var dataSlices []dataSlice
defer prepareDataForBigQuery(dataSlices, *ciFlag)
...
go func() {
  dataSlices = recordTransactionData(
    lg,
    sliceDuration,
    runStartTime,
    gitSha,
    goVersion,
    osVersion)
```

and return the slices from recordTransactionData when it exits

Reply:
or better yet, use a channel to pass the dataSlices back (my above sample code is not threadsafe)

Reply:
clever to use the channels to pass them back. I'll update to do that. Thanks.